prow/sidecar/run.go

```diff
@@ -208,17 +211,31 @@ func (o Options) preUpload() {
 	}
 }
 
-func (o Options) doUpload(ctx context.Context, spec *downwardapi.JobSpec, passed, aborted bool, metadata map[string]interface{}, logReaders map[string]io.Reader) error {
+func (o Options) doUpload(ctx context.Context, spec *downwardapi.JobSpec, passed, aborted bool, metadata map[string]interface{}, logReaders map[string]io.Reader, logFile *os.File) error {
 	startTime := time.Now()
 	logrus.Info("Starting to upload")
-	defer func() { logrus.WithField("duration", time.Since(startTime).String()).Info("Finished uploading") }()
-
 	uploadTargets := make(map[string]gcs.UploadFunc)
 
+	defer func() {
+		logrus.WithField("duration", time.Since(startTime).String()).Info("Finished uploading")
+	}()
+
 	for logName, reader := range logReaders {
 		uploadTargets[logName] = gcs.DataUpload(reader)
 	}
 
+	logFile.Sync()
+	logrus.SetOutput(os.Stderr)
+	logFileName := logFile.Name()
+	logFile.Close()
```

Comment:
We'll still incorrectly close `logFile` twice if `doUpload` is called twice (and we'll call `Sync` after closing the file). Let's only do these few lines the first time `doUpload` is called.